### ISSUE TITLE
Add function in Dataset that takes train/val/test splits and attaches negative samples

### ISSUE THREAD

**Issue by tizianocitro** [hypernetwork-research-group/hypertorch#398]
### Is your feature request related to a problem? Please describe.

Hyperlink-prediction workflows currently generate negative samples independently for the train, validation, and test datasets. When each sampler uses the same seed and transductive hypergraph context, validation and test negatives can overlap with training negatives.

Avoiding this overlap requires manually generating a single negative pool, partitioning its `HData`, rebasing hyperedge IDs, preserving attributes and masks, and attaching each partition to its corresponding dataset. This is too much internal data manipulation for examples and user code.

### Describe the solution you would like

Add a function to `Dataset` that accepts train, validation, and test splits and attaches disjoint negative samples to them.

The function should:

- Accept the datasets and the number or ratio of negatives required for each split.
- Generate one negative pool containing the total requested number of samples.
- Partition that pool into disjoint train, validation, and test subsets.
- Attach each partition to its corresponding dataset.
- Preserve node spaces, labels, hyperedge attributes, weights, and target masks.
- Reject negatives that match any positive hyperedge in the shared context.
- Be reproducible when given a seed.
- Validate that transductive splits share the same hypergraph context.
- Guarantee that negative hyperedges do not overlap across splits.
- Include unit tests covering counts, reproducibility, invalid inputs, and cross-split disjointness.

An example API could look like:

```python
train_dataset, val_dataset, test_dataset = Dataset.add_negative_samples_to_splits(
    datasets=[train_dataset, val_dataset, test_dataset],
    num_negative_samples=[
        len(train_dataset),
        len(val_dataset),
        int(len(test_dataset) * 0.6),
    ],
    num_nodes_per_sample=int(train_dataset.stats()["avg_degree_hyperedge"]),
    seed=42,
)
```

Hyperlink-prediction examples should use this function instead of independently sampling negatives for each split.

### Describe alternatives you've considered

- Using a different seed for each split. This avoids deterministic prefix duplication but does not guarantee disjoint negative sets.
- Manually generating one large negative pool and partitioning its `HData`. This requires handling internal hyperedge IDs, attributes, weights, labels, node spaces, and target masks in every caller.
- Keeping the existing independent sampling loops and explicitly removing overlaps afterward. This is less efficient and can leave a split with fewer negatives than requested.

### Additional context

The issue is especially visible with transductive splits because train, validation, and test contain the same complete `hyperedge_index` and differ primarily through `target_hyperedge_mask`. Using the same seed therefore produces the same accepted negative sequence in each independent sampler call.

Centralizing this operation in `Dataset` would provide a safe default and remove duplicated negative-partitioning logic from the hyperlink-prediction examples and benchmarks.
